### ISSUE TITLE
Negative losses when training with ntu

### ISSUE THREAD

**Issue by yyccli** [DegardinBruno/Kinetic-GAN#1]
Hi, first thank you for this great work!
I'm trying to training Kinetic-GAN using ntu dataset, all configurations are set by default, and i encountered negative losses when training. I'm not sure whether it's reasonable to get negative losses when training? Looking forward to your reply : )

p.s I'm using batch_size = 128 to accelerate training.

![kinetic-gan](https://user-images.githubusercontent.com/25741435/140271311-9c144e7c-b6f3-4e55-8c4d-77a899840a6d.png)

**Comment by DegardinBruno** [comment]
Hey @yyccli , thank you for your interest in Kinetic-GAN!
Yes, it is completely normal that the Wasserstein loss (and gradient penalty, of course) becomes negative in both the generator and critic. Wasserstein loss compares the mean of two distributions (line 152 and 171 of [kinetic-gan.py](https://github.com/DegardinBruno/Kinetic-GAN/blob/main/kinetic-gan.py)).
The loss becomes negative if one of the distributions is far apart from the other side of the other distribution.

Eventually, they will start to converge and losses will become stable!

**Comment by yyccli** [comment]
Hi, I have been training for 5+ days and i used the generated samples to visualize the actions. It's really amazing that the actions starting to look close to the real actions! Really Amazing Work!

**Comment by DegardinBruno** [comment]
Thanks for enjoying the codebase!
Despite some good results, there's still a lot of future points to improve, mainly due to some errors present in the training data.
For any additional information, please feel free to ask.
